doc/examples/filters/plot_deconvolution.py

```diff
@@ -19,23 +19,32 @@
 
 import numpy as np
 import matplotlib.pyplot as plt
+import skimage as ski
 
 from scipy.signal import convolve2d as conv2
 
-from skimage import color, data, restoration
 
 rng = np.random.default_rng()
 
-astro = color.rgb2gray(data.astronaut())
+# Convert astronaut image to grayscale
+astro = ski.color.rgb2gray(ski.data.astronaut())
 
+# Define PSF
 psf = np.ones((5, 5)) / 25
-astro = conv2(astro, psf, 'same')
-# Add Noise to Image
-astro_noisy = astro.copy()
-astro_noisy += (rng.poisson(lam=25, size=astro.shape) - 10) / 255.0
 
-# Restore Image using Richardson-Lucy algorithm
-deconvolved_RL = restoration.richardson_lucy(astro_noisy, psf, num_iter=30)
+# Convolve image with the PSF to simulate a blurred image
+astro_blurred = conv2(astro, psf, 'same')
+
+# Introduce poisson noise to the blurred image (https://en.wikipedia.org/wiki/Image_noise)
+max_photon_count = 100
```

Comment:
I know I originally suggested max_photon_count = 100, but on reflection I suggest using 1000 so that it more closely corresponds to the appearance of the previous output of the example (with incorrect noise model). 1000 displays some minor edge artefacts and noise amplification (as does the previous example code) but generally gives the intended impression of successful deblurring.